src/uct/ib/rc/base/rc_iface.c

```diff
@@ -787,11 +788,10 @@ ucs_status_t uct_rc_iface_qp_connect(uct_rc_iface_t *iface, struct ibv_qp *qp,
                                     IBV_QP_MIN_RNR_TIMER;
 
 #if HAVE_DECL_IBV_EXP_QP_OOO_RW_DATA_PLACEMENT
-    if (iface->config.ooo_rw &&
-        UCX_IB_DEV_IS_OOO_SUPPORTED(&uct_ib_iface_device(&iface->super)->dev_attr,
-                                    rc)) {
-        ucs_debug("enabling out-of-order on RC QP %x dev %s", qp->qp_num,
-                   uct_ib_device_name(uct_ib_iface_device(&iface->super)));
+    dev = uct_ib_iface_device(&iface->super);
+    if (iface->config.ooo_rw && UCX_IB_DEV_IS_OOO_SUPPORTED(dev, rc)) {
+        ucs_debug("enabling out-of-order on RC QP %x dev %s",
```

Comment:
can we reuse uct_dc_mlx5_iface_ooo_flag() ?

Reply:
Here we also rely on if HAVE_DECL_IBV_EXP_QP_OOO_RW_DATA_PLACEMENT to call either `ibv_exp_modify_qp` or `ibv_modify_qp`.
And in  uct_dc_mlx5_iface_ooo_flag() we check support for both DCT_OOO and QP_OOO, because we do not have to enable AR if only one of them is supported.  This is not relevant for RC, so it will be hard to use single generic function for all, or part of the uct_dc_mlx5_iface_ooo_flag() checks would need to be moved outside (increasing code duplication in DC).

Reply:
ok